src/components/StaticHeaderPageLayout.js

```diff
@@ -0,0 +1,88 @@
+import _ from 'underscore';
+import React, {useMemo} from 'react';
+import PropTypes from 'prop-types';
+import {ScrollView, View} from 'react-native';
+import headerWithBackButtonPropTypes from './HeaderWithBackButton/headerWithBackButtonPropTypes';
+import HeaderWithBackButton from './HeaderWithBackButton';
+import ScreenWrapper from './ScreenWrapper';
+import styles from '../styles/styles';
+import themeColors from '../styles/themes/default';
+import * as StyleUtils from '../styles/StyleUtils';
+import useWindowDimensions from '../hooks/useWindowDimensions';
+import FixedFooter from './FixedFooter';
+
+const propTypes = {
+    ...headerWithBackButtonPropTypes,
+
+    /** Children to display in the lower half of the page (below the header section w/ an animation) */
+    children: PropTypes.node.isRequired,
+
+    /** The background color to apply in the upper half of the screen. */
+    backgroundColor: PropTypes.string,
+
+    /** A fixed footer to display at the bottom of the page. */
+    footer: PropTypes.node,
+};
+
+const defaultProps = {
+    backgroundColor: themeColors.appBG,
+    footer: null,
+};
+
+function StaticHeaderPageLayout({backgroundColor, children, image: Image, footer, imageContainerStyle, style, ...propsToPassToHeader}) {
+    const {windowHeight} = useWindowDimensions();
+
+    const titleColor = useMemo(() => (backgroundColor === themeColors.appBG ? undefined : themeColors.textColorfulBackground), [backgroundColor]);
```

Comment:
const {titleColor, iconFill } = useMemo(() => {your logic here}) - you have the same check backgroundColor === themeColors.appBG - and both logic can be combined in one.